### PR TITLE
adding comparison operator

### PR DESCRIPTION
**Pull request by ianzapolsky**
as well as standard scala/haskell list accessors head, last, init, and tail

### CODE REVIEW

File: src/collections.h [hunk 1]
```diff
@@ -53,6 +53,18 @@ class Collection {
     T
     front();
 
+    T
+    head();
```
Comment (natebrennand) [inline src/collections.h]: I don't think we should have both `head` and `front`.

File: src/collections.h [hunk 2]
```diff
@@ -53,6 +53,18 @@ class Collection {
...
+
+    T
+    last();
+
+    Collection<T>
+    init();
```
Comment (natebrennand) [inline src/collections.h]: What is this function for?

**Comment by natebrennand** [comment]
this isn't mergeable currently FYI

File: src/collections.h (hunk 2 above)
Comment (ianzapolsky) [inline src/collections.h]: returns every element except for the last

Comment (natebrennand) [inline src/collections.h]: Is that a standard use case? I haven't seen that anywhere before.

Comment (ianzapolsky) [inline src/collections.h]: scala and haskell lists use it

**Pull request merged**